src/System.Collections.Specialized/src/System/Web/Util/HttpEncoder.cs

```diff
@@ -0,0 +1,782 @@
+                        }
+                        else
+                        {
+                            output.Write(ch);
+                        }
+                    }
+                }
+            }
+        }
+
+        internal static string HtmlDecode(string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return value;
+            }
+
+            return WebUtility.HtmlDecode(value);
+        }
+
+        internal static void HtmlDecode(string value, TextWriter output)
+        {
+            if (output == null)
+                throw new ArgumentNullException(nameof(output));
+            output.Write(WebUtility.HtmlDecode(value));
+        }
+
+        internal static string HtmlEncode(string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return value;
+            }
+            return WebUtility.HtmlEncode(value);
+        }
+
+        internal static void HtmlEncode(string value, TextWriter output)
+        {
+            if (output == null)
+                throw new ArgumentNullException(nameof(output));
+            output.Write(WebUtility.HtmlEncode(value));
+        }
+
+        private static unsafe int IndexOfHtmlAttributeEncodingChars(string s, int startPos)
+        {
+            Debug.Assert(0 <= startPos && startPos <= s.Length, "0 <= startPos && startPos <= s.Length");
+            int cch = s.Length - startPos;
+            fixed (char* str = s)
+            {
+                for (char* pch = &str[startPos]; cch > 0; pch++, cch--)
+                {
+                    char ch = *pch;
+                    if (ch <= '<')
+                    {
+                        switch (ch)
+                        {
+                            case '<':
+                            case '"':
+                            case '\'':
+                            case '&':
+                                return s.Length - cch;
+                        }
+                    }
+                }
+            }
+
+            return -1;
+        }
+
+        private static bool IsNonAsciiByte(byte b)
+        {
+            return (b >= 0x7F || b < 0x20);
+        }
+
+        internal static string JavaScriptStringEncode(string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return string.Empty;
+            }
+
+            StringBuilder b = null;
+            int startIndex = 0;
+            int count = 0;
+            for (int i = 0; i < value.Length; i++)
+            {
+                char c = value[i];
+
+                // Append the unhandled characters (that do not require special treament)
+                // to the string builder when special characters are detected.
+                if (CharRequiresJavaScriptEncoding(c))
+                {
+                    if (b == null)
+                    {
+                        b = new StringBuilder(value.Length + 5);
```

Comment:
>); [](start = 62, length = 2)

if you want to sprinkle magic perf dust over this you can do

            StringBuilder sb = StringBuilderCache.Acquire(value.Length + 5);
// use sb
            return StringBuilderCache.GetStringAndRelease(sb);